fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java

```diff
@@ -682,7 +682,8 @@ public List<HdfsFileDesc> getHdfsFileDescs(String dirPath, boolean isSplittable,
         // block locations of the files in the given path in one operation.
         // The performance is better than getting status and block location one by one.
         try {
-            RemoteIterator<LocatedFileStatus> blockIterator = fileSystem.listLocatedStatus(new Path(uri.getPath()));
+            // files in hdfs may have multiple directories, so we need to list all files in hdfs recursively here
+            RemoteIterator<LocatedFileStatus> blockIterator = fileSystem.listFiles(new Path(uri.getPath()), true);
```

Comment:
this is not a bug, it's a enhancement. this should be configurable.

Reply:
> this is not a bug, it's a enhancement. this should be configurable.

After consideration,it's more appropriate to put the para into `fe.conf`.